app/models/journal.rb

```diff
@@ -110,10 +110,14 @@ def noop?
   private
 
   def predecessor
-    @predecessor ||= self.class
-                     .where(journable_type: journable_type, journable_id: journable_id)
-                     .where("#{self.class.table_name}.version < ?", version)
-                     .order("#{self.class.table_name}.version DESC")
-                     .first
+    @predecessor ||= if initial?
```

Comment:
`nil` is not being memoized anyway, so this can just become `unless initial?` and omit the nil block